packages/files-ui/src/Components/Modules/FileBrowsers/Sharing/PermissionsDropdown.tsx

```diff
@@ -0,0 +1,141 @@
+
+import { MenuDropdown, IMenuItem } from "@chainsafe/common-components"
+import { createStyles, makeStyles } from "@chainsafe/common-theme"
+import { NonceResponsePermission } from "@chainsafe/files-api-client"
+import { t } from "@lingui/macro"
+import clsx from "clsx"
+import React, { ReactNode, useCallback } from "react"
+import { CSFTheme } from "../../../../Themes/types"
+
+const useStyles = makeStyles(
+  ({ constants, palette }: CSFTheme) => {
+    return createStyles({
+      icon: {
+        "& svg": {
+          fill: constants.header.iconColor
+        }
+      },
+      menuItem: {
+        width: "100%",
+        display: "flex",
+        flexDirection: "row",
+        alignItems: "center",
+        color: constants.header.menuItemTextColor,
+        "& svg": {
+          width: constants.generalUnit * 2,
+          height: constants.generalUnit * 2,
+          marginRight: constants.generalUnit,
+          fill: palette.additional["gray"][7],
+          stroke: palette.additional["gray"][7]
+        }
+      },
+      paddedTitle: {
+        paddingRight: constants.generalUnit * 2
+      },
+      permissionDropdownNoBorder: {
+        padding: `0px ${constants.generalUnit * 0.75}px`,
+        backgroundColor: palette.additional["gray"][1]
+      },
+      permissionDropDownBorders: {
+        border: `1px solid ${palette.additional["gray"][5]}`,
+        width: "inherit",
+        marginRight: constants.generalUnit,
+        borderRadius: "2px"
+      }
+    })
+  }
+)
+
+
+export interface LinkMenuItems {
+  id: NonceResponsePermission
+  onClick: () => void
+  contents: ReactNode
+}
+
+export const readRights = t`view-only`
+export const editRights = t`can-edit`
+export const translatedPermission = (permission: NonceResponsePermission) => permission === "read" ? readRights : editRights
+
+interface Props {
+  onViewPermissionClick?: () => void
+  onEditPermissionClick?: () => void
+  selectedPermission: NonceResponsePermission
+  withBorders: boolean
+  injectedClasses: {
+    root?: string
+    options?: string
+    dropdownTitle?: string
+  }
+  permissions: NonceResponsePermission[]
+  testId?: string
+}
+
+const PermissionsDropdown = ({
+  onViewPermissionClick,
+  onEditPermissionClick,
+  selectedPermission,
+  withBorders,
+  permissions,
+  injectedClasses,
+  testId
+}: Props) => {
+
+  const classes = useStyles()
+
+  const getMenuItems = useCallback(() => {
+    const menuItems: IMenuItem[] = []
+    if (permissions.includes("read")) {
+      menuItems.push(
+        {
+          onClick: onViewPermissionClick,
+          contents: (
+            <div
+              data-cy={testId && "menu-read"}
```

Comment:
This is not being passed a `testId` from anywhere so can just be like this:

```suggestion
              data-cy="menu-read"
```

Reply:
There is a testId passed in to Permission Dropdown [here](https://github.com/ChainSafe/ui-monorepo/blob/6b5756995235baf4b0dd4efe1ae685afe28e7d6d/packages/files-ui/src/Components/Modules/FileBrowsers/Sharing/LinkList.tsx#L218).

This is done to prevent duplicates when `cy.get` is called. The permission dropdown is reused in CreateOrManageSharedFolder.

Reply:
IIUC, the problem is that the same `PermissionDropdown` is used twice on the same modal, is that right? Then I'd build it as if it was an external component. So we use a `testId` prop, like we do here, but I'd change the `data-cy` to something like this:
 ```suggestion
              data-testid={`dropdown-permission-${testId}`}
```

Reply:
Yeah that's exactly the issue. Will implement a fix as described.